torch/futures/__init__.py

```diff
@@ -81,6 +81,39 @@ def then(self, callback):  # type: (Callable[[Future[T]], S]) -> Future[S]
         """
         return cast(Future[S], super().then(callback))
 
+    # Have to use string annotations because  PEP-0563 is not available in 3.6
+    def add_done_callback(self, callback):  # type: (Callable[[Future[T]], None]) -> None
+        r"""
+        Add a callback to this ``Future``.
+        Multiple callbacks can be added to the same ``Future``, and all
+        added callbacks will be executed once the future completes.
+        If the future has already completed, the callback will be executed immediately.
+
+        Append the given callback function to this ``Future``, which will be run
```

Comment:
There are some duplication between the two paragraphs. Is this intentional?

Reply:
fixed, thanks :/